### PR TITLE
EGM ID Updates from recipes (76X)

### PR DESCRIPTION
**Pull request by lgray**
This PR is brings CMSSW in line with the latest EGM recipes.

New IDs: 
- Triggering electron MVA ID for 25ns and 50ns
- V2 of cutbased electron ID for Spring15, HLT safe at "tight" working point

Bugfix:
- Photon cutbased ID was applying linear scaling in the endcaps for neutral had iso, it should have been exponential.

Externals PR is here: https://github.com/cms-data/RecoEgamma-ElectronIdentification/pull/6
Please wait until this is in an IB before testing.

@ikrav

### CODE REVIEW

**Comment by slava77** [comment]
@lgray here and in your other two PRs I'm not going to trigger tests, assuming you will detect when the IB with the externals show up

**Comment by lgray** [comment]
@slava77 Yes, I'll take care of starting the tests.

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
Errors seem to be with runTheMatrix.py itself:
```
Deprecated, please use -l limited
processing relval_standard
ERROR in Step
overwritting RunHLTPhy2015C not allowed
```
and then it quits...

Will wait for the IBs to be fixed before proceeding.
Luckily VID unit tests are OK!

**Comment by smuzaffar** [comment]
please test

**Comment by lgray** [comment]
@cmsbuild please test

File: RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Spring15Trig.cc [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+  allMVAVars.gsfhits         = eleRecoPtr->gsfTrack()->hitPattern().trackerLayersWithMeasurement();
+  allMVAVars.expectedMissingInnerHits = eleRecoPtr->gsfTrack()
+    ->hitPattern().numberOfHits(reco::HitPattern::MISSING_INNER_HITS);
+
+  reco::ConversionRef conv_ref = ConversionTools::matchedConversion(*eleRecoPtr,
+								    conversions, 
+								    theBeamSpot->position());
+  double vertexFitProbability = -1.; 
+  if(!conv_ref.isNull()) {
+    const reco::Vertex &vtx = conv_ref.get()->conversionVertex(); if (vtx.isValid()) {
```
Comment (slava77) [inline RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Spring15Trig.cc]: new line in front of the "if" would be nice

File: RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndExpoScalingCut.cc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  float anyisoval = -1.0;
+  if( _anyPFIsoMap.isValid() && _anyPFIsoMap->contains( cand.id() ) ) {
+    anyisoval = (*_anyPFIsoMap)[cand];
+  } else if ( _anyPFIsoMap.isValid() && _anyPFIsoMap->idSize() == 1 &&
+              cand.id() == edm::ProductID() ) {
+    // in case we have spoofed a ptr
+    //note this must be a 1:1 valuemap (only one product input)
+    anyisoval = _anyPFIsoMap->begin()[cand.key()];
+  } else if ( _anyPFIsoMap.isValid() ){ // throw an exception
+    anyisoval = (*_anyPFIsoMap)[cand];
```
Comment (slava77) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndExpoScalingCut.cc]: maybe a more explicit exception message would be more useful?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/11738	 2f1cf60
- code changes are in line with the PR description
    - minor code review comments can be addressed separately
- jenkins tests pass and comparisons with baseline show no differences (the only changes expected are in ID variables [user floats ] in the slimmedPhotons and electrons)
- local tests show
    - changes in timing of electronMVAValueMapProducer increasing by almost x2 (still under 1ms/evt): expected since new variables are computed for electrons, and the same count with similar complexity for photons
    - event size changes show differences only in miniAOD, most visible in the slimmedElectrons (1% increase in the branch size on 200 events from DoubleEG PD)

**Pull request merged**